onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -272,31 +272,42 @@ def aten_angle(self: TensorType) -> TensorType:
     raise NotImplementedError()
 
 
-@torch_op("aten::any", trace_only=True)
-def aten_any(self: TTensor, dim: Optional[int] = None, keepdim: bool = True) -> BOOL:
+@torch_op("aten::any")
+def aten_any(
+    self: TTensor,
+    keepdim: bool = True,  # pylint: disable=unused-argument
+) -> BOOL:
     """any(Tensor self) -> Tensor"""
 
-    negative_one = op.Constant(value_ints=[-1])
     self_rank = op.Size(op.Shape(self))
     if self_rank == 0:
-        self = op.Reshape(self, negative_one)
-
-    # cannot cast to INT64 because 0.1 will be cast to 0, then convert to false
-    self_bool = op.Cast(self, to=BOOL.dtype)
-    # op.ReduceMax() in next step cannot calculate BOOL value, so convert to INT64
-    self_int = op.Cast(self_bool, to=INT64.dtype)
-
-    if op.OptionalHasElement(dim):
-        dim = op.Reshape(dim, negative_one)
-        dims = op.Cast(dim, to=INT64.dtype)
-        result_max = op.ReduceMax(self_int, dims, keepdims=keepdim, noop_with_empty_axes=0)
+        result = op.Not(op.Equal(self, 0.0))
     else:
+        # cannot cast to INT64 because 0.1 will be cast to 0, then convert to false
+        self_bool = op.Cast(self, to=BOOL.dtype)
+        # op.ReduceMax() in next step cannot calculate BOOL value, so convert to INT64
+        self_int = op.Cast(self_bool, to=INT64.dtype)
         result_max = op.ReduceMax(self_int, keepdims=0, noop_with_empty_axes=0)
+        result = op.Greater(result_max, op.Constant(value_int=0))
+    return result
 
-    result = op.Greater(result_max, op.Constant(value_int=0))
-    if self_rank == 0:
-        result = op.Squeeze(result)
 
+@torch_op("aten::any", overload=True)
+def aten_any_dim(self: TTensor, dim: int, keepdim: bool = True) -> BOOL:
+    """any(Tensor self) -> Tensor"""
+
+    self_rank = op.Size(op.Shape(self))
+    if self_rank == 0:
+        result = op.Not(op.Equal(self, 0.0))
+    else:
+        # cannot cast to INT64 because 0.1 will be cast to 0, then convert to false
+        self_bool = op.Cast(self, to=BOOL.dtype)
+        # op.ReduceMax() in next step cannot calculate BOOL value, so convert to INT64
+        self_int = op.Cast(self_bool, to=INT64.dtype)
+        # Change int(dim) to tensor(dims)
```

Comment:
More precisely is it changing dim from INT64 to INT64[1]?

Reply:
from int to INT64[1]